packages/addons-v5/lib/addons_wait.js

```diff
@@ -1,26 +1,60 @@
+/* eslint-disable no-await-in-loop */
+
 'use strict'
 
 const cli = require('heroku-cli-util')
 
-module.exports = async function (api, addon, interval) {
-  const app = addon.app.name
-  const addonName = addon.name
+module.exports = {
+  waitForAddonProvisioning: async function (api, addon, interval) {
+    const app = addon.app.name
+    const addonName = addon.name
+
+    await cli.action(`Creating ${cli.color.addon(addon.name)}`, (async function () {
+      while (addon.state === 'provisioning') {
```

Comment:
I think this would be easier to understand and potentially safer with a `setInterval` that will eventually call `resolve` from a Promise. Something like
```
const myPromise = new Promise((resolve, reject) => {
  const intervalId = setInterval(() => {
    if (thing) {
      clearInterval(intervalId);
      resolve("foo");
    }
  }, 1000);
});

await myPromise
```

Reply:
It indeed gave me some headaches to understand what was that for. After looking a little bit I discovered that this seems to be a very well known and standard way of implementing a `sleep()` function in Javascript. The `while` loop and this sleep function implement some rudimentary polling method, to avoid making a zillion requests to the API to see if the process was completed. I believe it's correct in this context, but I'm not sure how to change it to keep the behavior unchanged: wait `interval` seconds and repeat the API request to fetch the add-on again.

Reply:
I didn't read the PR closely enough and thought this was a new convention, not repurposed existing code.